pkg/validate/dynamic/platformworkloadidentityprofile.go

```diff
@@ -0,0 +1,108 @@
+package dynamic
+
+import (
+	"context"
+	"net/http"
+
+	sdkauthorization "github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/authorization/armauthorization/v3"
+	"github.com/Azure/go-autorest/autorest/azure"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/azuresdk/armauthorization"
+	"github.com/Azure/ARO-RP/pkg/util/rbac"
+	"github.com/Azure/ARO-RP/pkg/util/stringutils"
+)
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+func (dv *dynamic) ValidatePlatformWorkloadIdentityProfile(ctx context.Context, oc *api.OpenShiftCluster, platformWorkloadIdentityRoles []api.PlatformWorkloadIdentityRole, roleDefinitions armauthorization.RoleDefinitionsClient) error {
+	dv.log.Print("ValidatePlatformWorkloadIdentityProfile")
+
+	platformIdentitiesActionsMap := map[string][]string{}
+
+	requiredOperatorIdentities := []string{}
+	recievedOperatorIdentities := []string{}
+	for _, role := range platformWorkloadIdentityRoles {
+		requiredOperatorIdentities = append(requiredOperatorIdentities, role.OperatorName)
+	}
+	for _, role := range oc.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities {
+		recievedOperatorIdentities = append(recievedOperatorIdentities, role.OperatorName)
+	}
+	ok := stringutils.ElementsMatch(requiredOperatorIdentities, recievedOperatorIdentities)
+	if !ok {
+		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodePlatformWorkloadIdentityMismatch,
+			"properties.ValidatePlatformWorkloadIdentityProfile.PlatformWorkloadIdentities", "There's a mismatch between the required and expected set of platform workload identities for the requested OpenShift version '%s'", oc.Properties.ClusterProfile.Version)
+	}
+
+	err := dv.validateClusterMSI(ctx, oc)
+	if err != nil {
+		return err
+	}
+
+	for _, role := range platformWorkloadIdentityRoles {
+		definition, err := dv.roleDefinitions.GetByID(ctx, role.RoleDefinitionID, &sdkauthorization.RoleDefinitionsClientGetByIDOptions{})
+		if err != nil {
+			return err
+		}
+
+		if len(definition.Properties.Permissions) <= 0 {
+			return api.NewCloudError(http.StatusInternalServerError, api.CloudErrorCodeInvalidClusterMSICount,
+				"dynamic.ValidatePlatformWorkloadIdentityProfile", "No Permissions exists for the role %s", role.RoleDefinitionID)
+		}
+
+		actions := []string{}
+		for _, action := range definition.Properties.Permissions[0].Actions {
+			actions = append(actions, *action)
+		}
+		platformIdentitiesActionsMap[role.OperatorName] = actions
```

Comment:
This code is duplicated in `validateClusterMSIPermissions()` - should we make a new shared function for both?

Reply:
Added a shared function for the same.